### PR TITLE
Package Splitting works 

### PR DESCRIPTION
**Pull request by lukasjhan**
I've done the initial version of splitting work.

the package spitted by

- core: issue, present, verify
- decode: decode sdjwt into object
- hash: purejs sha256 function
- present: present sdjwt & get presentable keys
- utils
- types
- node-crypto: contain sample crypto logic for optional use for nodejs
- browser-crypto: contain sample crypto logic for optional use for browser

Todos:

- support sync functions for decode, present and core(issue, present, verify)
- fix examples

### CODE REVIEW

**Comment by lukasjhan** [comment]
I'll add the sd-jwt-vc in next PR

**Comment by cre8** [comment]
@lukasjhan I added some notes, most of them are relevant for all packages since it is based on the package json.

I am also missing code documentation. Even when the functions are small it is good for a quality based project ;)

**Comment by lukasjhan** [comment]
@cre8 Good points :) 
I'll make a docs right away

**Comment by lukasjhan** [comment]
I should have written some comments while I was working on it.

**Comment by lukasjhan** [comment]
> @lukasjhan I added some notes, most of them are relevant for all packages since it is based on the package json.
> 
> I am also missing code documentation. Even when the functions are small it is good for a quality based project ;)

Sorry for ping again.
Where can I find you notes? I couldn't find it.

File: .github/workflows/test-pr.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Install pnpm
         run: npm install -g pnpm
       - name: Install Dependencies
-        run: pnpm install
+        run: pnpm install && pnpm build && pnpm install
```
Comment (lukasjhan) [inline .github/workflows/test-pr.yaml:20]: Since we use pnpm workspace, I think it need to be built before install

**Review by zustkeeper — DISMISSED** [review]

**Comment by lukasjhan** [comment]
I added

- comments
- examples
- README of examples

**Comment by zustkeeper** [comment]
> I added
> 
> * comments
> * examples
> * README of examples

I checked the examples and it works well!

**Review by zustkeeper — APPROVED** [review]

**Comment by cre8** [comment]
@lukasjhan With code documentation I do not mean some google docs explaining implementation decision. But documentation for variables, functions, classes, etc directly in the code

**Comment by lukasjhan** [comment]
> @lukasjhan With code documentation I do not mean some google docs explaining implementation decision. But documentation for variables, functions, classes, etc directly in the code

I got it. I wrote some comments in the code. you can check here :) [a50d838](https://github.com/openwallet-foundation-labs/sd-jwt-js/pull/68/commits/a50d8385be590585fb21578a0fc207a58dd89ebc)

**Comment by cre8** [comment]
> > @lukasjhan With code documentation I do not mean some google docs explaining implementation decision. But documentation for variables, functions, classes, etc directly in the code
> 
> I got it. I wrote some comments in the code. you can check here :) [a50d838](https://github.com/openwallet-foundation-labs/sd-jwt-js/pull/68/commits/a50d8385be590585fb21578a0fc207a58dd89ebc)

Great. Do you think it makes sense to also add eslint during this PR? opened an issue for this, see #69

**Comment by lukasjhan** [comment]
I would like to merge this PR and do other task in different PR. It's too big :)

**Pull request merged**